### PR TITLE
Add openff-fragmenter

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by SimonBoothroyd** [comment]
I think the windows CI is failing due to me not including a .bat install script, but I'm not sure this is needed here as it is a noarch package? 

There may also be windows failures due to a missing downstream dependency on windows (`ambertools`) however this shouldn't be an issue as windows is not a priority for this package.

**Comment by SimonBoothroyd** [comment]
@conda-forge/help-python this should be ready for review!

File: recipes/openff-fragmenter/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+source:
+  url: https://github.com/openforcefield/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 685256e0885ef23df7c4e8aa601a768fcb54867712671cf7d210d7f5284be7be
+
+build:
+  number: 0
+
+outputs:
+  - name: {{ name|lower }}-base
+    script: build_base.sh
```
Comment (chrisburr) [inline recipes/openff-fragmenter/meta.yaml:17]:
```suggestion
    script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/openff-fragmenter/build_base.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+${PYTHON} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
```
Comment (chrisburr) [inline recipes/openff-fragmenter/build_base.sh:1]: This file can be removed

File: recipes/openff-fragmenter/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{% set name = "openff-fragmenter" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}-recipe
```
Comment (chrisburr) [inline recipes/openff-fragmenter/meta.yaml:5]:
```suggestion
  name: {{ name|lower }}
```

**Comment by SimonBoothroyd** [comment]
Thanks for the review @chrisburr! I applied your suggestions in d5fdb53, however it seems that the CI is now failing due to [`PYTHON` not being found](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=324679&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=514) by the Jinja template. 

For split packages like this I thought a `build_base.sh` is required to resolve this?

File: recipes/openff-fragmenter/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        - python >=3.6
+      run:
+        - python >=3.6
+        - rdkit
+        - ambertools
+        - {{ pin_subpackage('openff-fragmenter-base', exact=True) }}
+
+    test:
+      imports:
+        - openff.fragmenter
```
Comment (chrisburr) [inline recipes/openff-fragmenter/meta.yaml:53]:
```suggestion
  - name: {{ name|lower }}
```

File: recipes/openff-fragmenter/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/openforcefield/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 685256e0885ef23df7c4e8aa601a768fcb54867712671cf7d210d7f5284be7be
+
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/openff-fragmenter/meta.yaml:13]:
```suggestion
  number: 0
  noarch: python

requirements:
  host:
    - python >=3.6
  run:
    - python >=3.6
    - rdkit
    - ambertools
    - {{ pin_subpackage('openff-fragmenter-base', exact=True) }}

test:
  imports:
    - openff.fragmenter
```

**Review by chrisburr — COMMENTED** [review]
I always forget how to do this correctly, I think this is it.

**Comment by SimonBoothroyd** [comment]
Hmm I tried the suggested changes in 21ac9dc but now it seems to be trying to copy over the install command

**Comment by mattwthompson** [comment]
I ran into some of these issues last November working on another package. I remember being pretty confused and somewhat frustrated [1]. What I was told then [2] was that there's a bug somewhere in the conda infrastructure that prevents `build: script: "{{ PYTHON }} -m pip install . -vv"` to work inside of an `outputs:` block, so the build script was needed (even though without `outputs:` it's not). Sorry I can't offer more of an explanation; I was able to get a feedstock building and didn't dig too deeply into the internals of why that syntax didn't work. This may have been changed since then; I wasn't able to find an issue on it so I didn't keep tabs on it.  I _think_ the recipe a few commits earlier (89b12cbda330f98e5338c924a0bbb70c11f23b48) should work - even though it's not ideal to have the build script split out it seems to be necessary.

I'm also not familiar with the strategy of building one package at the high-level `build:` and then having a second one in the `outputs:`. Is there a particular benefit of using that? The last time [3] we came across this I shoved both packages into the `outputs:` and it's been working fine. I could be missing something.


1. If I/we could concretely and accurately understand the details, I'd be happy to make an effort to improve the docs on this topic (https://github.com/conda-forge/conda-forge.github.io/issues/746).
2. https://gitter.im/conda-forge/conda-forge.github.io?at=5fb681fec10273610a15d99a
3. https://github.com/conda-forge/openff-toolkit-feedstock/blob/c08b5a84c677166ba845d2c433ba989b89b70668/recipe/meta.yaml)

**Comment by SimonBoothroyd** [comment]
@chrisburr ( / @conda-forge/help-python ) I've reverted the recipe back to not split the package at all to hopefully simplify things - this should be good for re-review!

**Pull request merged**